axelor-account/src/main/java/com/axelor/apps/account/service/moveline/massentry/MoveLineMassEntryGroupServiceImpl.java

```diff
@@ -232,8 +232,187 @@ public Map<String, Map<String, Object>> getAccountOnChangeAttrsMap(
+      throws AxelorException {
+    Map<String, Object> valuesMap = new HashMap<>();
+
+    if (moveLine.getPartner() == null) {
+      moveLineMassEntryRecordService.resetPartner(moveLine, null);
+    } else {
+      if (move != null && move.getJournal() != null) {
+        int journalTechnicalTypeSelect =
+            move.getJournal().getJournalType().getTechnicalTypeSelect();
+        moveLineMassEntryRecordService.setMovePaymentMode(moveLine, journalTechnicalTypeSelect);
+        move.setPartner(moveLine.getPartner());
+        moveLineMassEntryRecordService.setMovePaymentCondition(
+            moveLine, journalTechnicalTypeSelect);
+        moveLineMassEntryRecordService.loadAccountInformation(move, moveLine);
+        move.setPartner(null);
+        moveLineMassEntryRecordService.setVatSystemSelect(moveLine, move);
+      }
+      moveLineMassEntryRecordService.setMovePartnerBankDetails(moveLine);
+      moveLineMassEntryRecordService.setCurrencyCode(moveLine);
+    }
+
+    if (move != null) {
+      moveLineMassEntryRecordService.setMovePfpValidatorUser(moveLine, move.getCompany());
+      valuesMap.put("movePfpValidatorUser", moveLine.getMovePfpValidatorUser());
+    }
+
+    valuesMap.put("isEdited", MoveLineMassEntryRepository.MASS_ENTRY_IS_EDITED_ALL);
+    valuesMap.put("partner", moveLine.getPartner());
+    valuesMap.put("partnerId", moveLine.getPartnerId());
+    valuesMap.put("partnerSeq", moveLine.getPartnerSeq());
+    valuesMap.put("partnerFullName", moveLine.getPartnerFullName());
+    valuesMap.put("movePartnerBankDetails", moveLine.getMovePartnerBankDetails());
+    valuesMap.put("vatSystemSelect", moveLine.getVatSystemSelect());
+    valuesMap.put("taxLine", moveLine.getTaxLine());
+    valuesMap.put("analyticDistributionTemplate", moveLine.getAnalyticDistributionTemplate());
+    valuesMap.put("currencyCode", moveLine.getCurrencyCode());
+    valuesMap.put("movePaymentMode", moveLine.getMovePaymentMode());
+    valuesMap.put("movePaymentCondition", moveLine.getMovePaymentCondition());
+    valuesMap.put("account", moveLine.getAccount());
+
+    return valuesMap;
+  }
+
+  @Override
+  public Map<String, Map<String, Object>> getPartnerOnChangeAttrsMap(MoveLineMassEntry moveLine) {
+    Map<String, Map<String, Object>> attrsMap = new HashMap<>();
+
+    moveLineMassEntryAttrsService.addMovePfpValidatorUserReadOnly(moveLine, attrsMap);
+
+    return attrsMap;
+  }
+
+  @Override
+  public Map<String, Object> getInputActionOnChangeValuesMap(
+      MoveLineMassEntry moveLine, Move move) {
+    Map<String, Object> valuesMap = new HashMap<>();
+
+    if (moveLine.getInputAction() == MoveLineMassEntryRepository.MASS_ENTRY_INPUT_ACTION_MOVE) {
+      moveLine = new MoveLineMassEntry();
+      massEntryService.resetMoveLineMassEntry(moveLine);
+      moveLineMassEntryRecordService.setNextTemporaryMoveNumber(moveLine, move);
+      moveLine.setCounter(1);
+
+      valuesMap = this.setAllMoveLineValuesMap(moveLine);
+    }
```

Comment:
Same as before.
Also there should be only one method in the entire application that creates objects. Here it should call a method called `createMoveLineMassEntry` or something